pkg/lockfile/parse-go-lock_test.go

```diff
@@ -132,7 +132,7 @@ func TestParseGoLock_TwoPackages(t *testing.T) {
 		},
 		{
 			Name:      "stdlib",
-			Version:   "1.17.0",
+			Version:   "1.17.9999",
```

Comment:
Hmm, if we surface this to users, this might look a bit weird and raise a lot of questions.

I wonder if we should just  assume .0 here  to avoid false negatives as well. To mitigate false positives, users could either:

1. Just bump the Go version to another minor version.  This should be fairly "easy" to do for most users given that Go is backwards compatible?
2. Add an exclusion via an OSV toml config.

In  any case,  this problem will go away after some time as newer Go versions are adopted.  

This does have another implication though: Go library developers may be incentivised to have a lower go directive in go.mod, in maximise the compatibility of their library (since after go 1.21, the go toolchain will refuse to use modules declaring newer versions). Generally speaking, library developers may still need a way to turn the stdlib scanning off some other way, but I'm not sure what that should look like. Perhaps we can wait to see if we get any feedback on this.

Reply:
> 1. Just bump the Go version to another minor version. This should be fairly "easy" to do for most users given that Go is backwards compatible?

Even the latest go minor version (1.21.0) has 10+ vulnerabilities that's fixed over 1.21.1 - 1.21.5, 

> Add an exclusion via an OSV toml config.
Adding individual vulnerability exclusions in OSV toml config is not practical because of the sheer number of stdlib vulnerabilities. I do remember talking about adding package exclusions as an option to osv-scanner.toml. This will turn off all stdlib scanning though, even for out of support versions..

Reply:
I can make a better fix than adding `9999` though, it's just the easiest/cleanest way to implement this.
I'll just "inject" the patch version when sending to osv.dev API,

Reply:
Sgtm! Given the volume of vulns, injecting this patch version seems the most reasonable.